File: Languages/npeg_csharp/RobustHaven.Text.Npeg.Tests/Parsers/NpegNode.cs
using System;

namespace RobustHaven.Text.Npeg.Tests.Parsers
{
	public class NpegNode : NpegParser
	{
		public NpegNode(InputIterator iterator) : base(iterator)
		{
		}

		public override Boolean IsMatch()
		{
			return NpegNode_impl_0();
		}


		protected Boolean NpegNode_impl_0()
		{
			String _nodeName_0 = "NpegNode";
			return CapturingGroup(NpegNode_impl_1, _nodeName_0, false, false);
		}

		protected Boolean NpegNode_impl_1()
		{
			return PrioritizedChoice(NpegNode_impl_2, NpegNode_impl_3);
		}

		protected Boolean NpegNode_impl_3()
		{
			String _literal_0 = ".NET Parsing Expression Grammar";
			return Literal(_literal_0, false);
		}

		protected Boolean NpegNode_impl_2()
		{
			String _literal_0 = "NPEG";
			return Literal(_literal_0, false);
		}
	}
}

File: Languages/npeg_c/robusthaven.tests/ParserTest_NpegNode/NpegNode.c
#include <stdlib.h>
#include "NpegNode.h"

int NpegNode_impl_0(npeg_inputiterator* iterator, npeg_context* context)
{
	char* _nodeName_0 = "NpegNode";

	return npeg_CapturingGroup(iterator,  context, &NpegNode_impl_1, _nodeName_0, 0, 0);
}
int NpegNode_impl_1(npeg_inputiterator* iterator, npeg_context* context)
{
	return npeg_PrioritizedChoice(iterator,  context, &NpegNode_impl_2, &NpegNode_impl_3);
}
int NpegNode_impl_3(npeg_inputiterator* iterator, npeg_context* context)
{
	char* _literal_0 = ".NET Parsing Expression Grammar";

	return npeg_Literal(iterator,  context, _literal_0, 31, 0);
}
int NpegNode_impl_2(npeg_inputiterator* iterator, npeg_context* context)
{
	char* _literal_0 = "NPEG";

	return npeg_Literal(iterator,  context, _literal_0, 4, 0);
}
